gemPush failed even $HOME/.gem/credential setting properly.

`gradlew gemPush` failed even $HOME/.gem/credential setting properly.

MRI gem worked properly even no setting changed.
(Ruby 3.2.2, gem 3.4.19)

I think It relates to JRuby issue.


### environment

* OS: macOS 13.5.2 (M1 mac)
* Java: Zulu 1.8.0_362

### log

```
ls -l /Users/user/.gem/credentials
-rw-------@ 1 user  staff  81 Sep 14 13:40 /Users/user/.gem/credentials
```

```
./gradlew gemPush
Starting a Gradle Daemon (subsequent builds will be faster)
> Task :compileJava UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar UP-TO-DATE
> Task :gem UP-TO-DATE

> Task :gemPush
Executing: `java org.jruby.Main -rjars/setup -S gem push /path/to/plugin/embulk-parser-jsonpath/build/gems/embulk-parser-jsonpath-0.4.1-java.gem --verbose`
    with working directory at: /path/to/plugin/embulk-parser-jsonpath/build/gems
    with classpath: [ /Users/user/.gradle/caches/modules-2/files-2.1/org.jruby/jruby-complete/9.2.7.0/308120bca38f617e7b275af8ce0cbd9f0be66218/jruby-complete-9.2.7.0.jar ]
ERROR:  Your gem push credentials file located at:

	/Users/user/.gem/credentials

has file permissions of 0644 but 0600 is required.

To fix this error run:

	chmod 0600 /Users/user/.gem/credentials

You should reset your credentials at:

	https://rubygems.org/profile/edit

if you believe they were disclosed to a third party.

> Task :gemPush FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':gemPush'.
> Process 'command '/Library/Java/JavaVirtualMachines/zulu-8.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.8.2/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 6s

4 actionable tasks: 1 executed, 3 up-to-date
```

[log2.txt](https://github.com/embulk/gradle-embulk-plugins/files/12614471/log2.txt)

@dmikurube  Thank you for your reply

It caused `Could not set unknown property 'jruby' for task ':gemPush' of type org.embulk.gradle.embulk_plugins.GemPush.` What is missing?

```java
plugins {
    id "java"
    id "checkstyle"
    id "maven-publish"
    id "org.embulk.embulk-plugins" version "0.5.5"
    id "signing"
}
// ...
gemPush {
//    host = "https://rubygems.org"
    host = "http://localhost:9292"
    jruby = "org.jruby:jruby-complete:9.2.21.0"
}
```

Do I need update gradle?

```
./gradlew gemPush

FAILURE: Build failed with an exception.

* Where:
Build file '/path/to/embulk-parser-jsonpath/build.gradle' line: 111

* What went wrong:
A problem occurred evaluating root project 'embulk-parser-jsonpath'.
> Could not set unknown property 'jruby' for task ':gemPush' of type org.embulk.gradle.embulk_plugins.GemPush.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.8.2/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 594ms
```

9.2.21.0 may resolve this issue.

But, the current implementation have another issue. It prints `You have enabled multi-factor authentication. Please enter OTP code.`. But it doesn't show OTP prompt. As a result, `gemPush` failed.


### check code

```diff

Executing: `java org.jruby.Main -rjars/setup -S gem push /path/to/embulk-filter-calc/build/gems/embulk-filter-calc-0.2.1.pretest-java.gediff --git a/build.gradle b/build.gradle
index fb41948..f03ff72 100644
--- a/build.gradle
+++ b/build.gradle
@@ -4,14 +4,14 @@ plugins {
     id "antlr"
     id "maven-publish"
     id "signing"
-    id "org.embulk.embulk-plugins" version "0.5.5"
+    id "org.embulk.embulk-plugins" version "0.6.0"
 }
 repositories {
     mavenCentral()
 }

 group = "io.github.hiroyuki-sato"
-version = "0.2.1-SNAPSHOT"
+version = "0.2.1.pretest"
 description = "Caculator filter plugin for Embulk"


@@ -189,6 +189,7 @@ gem {

 gemPush {
     host = "https://rubygems.org"
+    jruby = "org.jruby:jruby-complete:9.2.21.0"
 }

 tasks.withType(JavaCompile) {
```

### Current behavior (JRuby 9.2.7.0)

```
./gradlew gemPush

> Task :gemPush
Executing: `java org.jruby.Main -rjars/setup -S gem push /path/to/embulk-filter-calc/build/gems/embulk-filter-calc-0.2.1.pretest-java.gem --verbose`
    with working directory at: /path/to/embulk-filter-calc/build/gems
    with classpath: [ /Users/user/.gradle/caches/modules-2/files-2.1/org.jruby/jruby-complete/9.2.7.0/308120bca38f617e7b275af8ce0cbd9f0be66218/jruby-complete-9.2.7.0.jar ]
ERROR:  Your gem push credentials file located at:

        /Users/user/.gem/credentials

has file permissions of 0644 but 0600 is required.

To fix this error run:

        chmod 0600 /Users/user/.gem/credentials

You should reset your credentials at:

        https://rubygems.org/profile/edit

if you believe they were disclosed to a third party.

> Task :gemPush FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':gemPush'.
> Process 'command '/Library/Java/JavaVirtualMachines/zulu-8.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.6.2/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 3s
5 actionable tasks: 1 executed, 4 up-to-date
```

### 9.2.21.0

```
./gradlew gemPush

> Task :gemPush
Executing: `java org.jruby.Main -rjars/setup -S gem push /path/to/embulk-filter-calc/build/gems/embulk-filter-calc-0.2.1.pretest-java.gem --verbose`
    with working directory at: /path/to/embulk-filter-calc/build/gems
    with classpath: [ /Users/user/.gradle/caches/modules-2/files-2.1/org.jruby/jruby-complete/9.2.21.0/d7be7522679da0a3f2aad68f8526927456ce906/jruby-complete-9.2.21.0.jar ]
GET https://rubygems.org/latest_specs.4.8.gz
304 Not Modified
Pushing gem to https://rubygems.org...
POST https://rubygems.org/api/v1/gems
401 Unauthorized
You have enabled multi-factor authentication. Please enter OTP code.
POST https://rubygems.org/api/v1/gems
401 Unauthorized
You have enabled multifactor authentication but no OTP code provided. Please fill it and retry.

> Task :gemPush FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':gemPush'.
> Process 'command '/Library/Java/JavaVirtualMachines/zulu-8.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.6.2/userguide/command_line_interface.html#sec:command_line_warnings

```

--- a/src/main/java/org/embulk/gradle/embulk_plugins/EmbulkPluginsPlugin.java
+++ b/src/main/java/org/embulk/gradle/embulk_plugins/EmbulkPluginsPlugin.java
@@ -290,6 +290,4 @@ public class EmbulkPluginsPlugin implements Plugin<Project> {
             return mavenVersion;
         }
     }
-
-    static final String DEFAULT_JRUBY = "org.jruby:jruby-complete:9.2.7.0";
 }

--- a/src/main/java/org/embulk/gradle/embulk_plugins/Gem.java
+++ b/src/main/java/org/embulk/gradle/embulk_plugins/Gem.java
@@ -106,7 +106,7 @@ class Gem extends AbstractArchiveTask {
         this.generateGemspec.set(true);
 
         this.jruby = objectFactory.property(Object.class);
-        this.jruby.set(EmbulkPluginsPlugin.DEFAULT_JRUBY);
+        this.jruby.set(DEFAULT_JRUBY);
 
         this.getArchiveExtension().set("gem");
     }
@@ -489,6 +489,8 @@ class Gem extends AbstractArchiveTask {
         builder.append(pathElementStream.map(pathElement -> pathElement.toString()).collect(Collectors.joining("/")));
         return builder.toString();
     }
+
+    static final String DEFAULT_JRUBY = "org.jruby:jruby-complete:9.4.3.0";
 
     private final Property<String> embulkPluginMainClass;
     private final Property<String> embulkPluginCategory;

--- a/src/main/java/org/embulk/gradle/embulk_plugins/GemPush.java
+++ b/src/main/java/org/embulk/gradle/embulk_plugins/GemPush.java
@@ -70,7 +70,7 @@ abstract class GemPush extends DefaultTask {
         this.host = objectFactory.property(String.class);
 
         this.jruby = objectFactory.property(Object.class);
-        this.jruby.set(EmbulkPluginsPlugin.DEFAULT_JRUBY);
+        this.jruby.set(Gem.DEFAULT_JRUBY);
     }
 
     @Incremental

